Install.lua

```diff
@@ -1,175 +1,275 @@
-	New.Name = Name
-	New.Source = LoadURL(URL) or error("Unable to load script")
-	return New
+local function GiveSourceToScript(Link, Script)
+	DataSources[Script] = Link
+	Script.Source = GetAsync(Link)
 end
 
-local function GetDirectory(Parent, URL)
-	local DirectoryName = URL:gmatch("%w+\\")()
-	if DirectoryName then
-		DirectoryName = DirectoryName:sub(1, #DirectoryName-1)
+local function InstallRepo(Link, Directory, Parent, Routines, TypesSpecified)
+	local Value = #Routines + 1
+	Routines[Value] = false
+	local Generated, MainExists
 
-		local New = Parent:FindFirstChild(DirectoryName) or Instance.new("Folder", Parent)
-		New.Name = DirectoryName
-		return GetDirectory(New, URL:sub(#DirectoryName+2, #URL))
-	else
-		return Parent
+	local ScriptCount = 0
+	local Scripts = {}
+
+	local FolderCount = 0
+	local Folders = {}
+
+	for Link in GetAsync(Link):gmatch("<tr class=\"js%-navigation%-item\">.-<a href=\"(.-)\" class=\"js%-navigation%-open\".-</tr>") do
+		if Link:find("/[^/]+/[^/]+/tree") then
+			FolderCount = FolderCount + 1
+			Folders[FolderCount] = Link
+		elseif Link:find("/[^/]+/[^/]+/blob.+%.lua$") then			
+			local ScriptName, ScriptClass = Link:match("([%w-_%%]+)%.?(%l*)%.lua$")
+
+			if ScriptName:lower() ~= "install" then
+				if ScriptClass == "mod" or ScriptClass == "module" then TypesSpecified = true end
+
+				if ScriptName == "_" or ScriptName:lower() == "main" then
+					ScriptCount = ScriptCount + 1
+					for a = ScriptCount, 2, -1 do
+						Scripts[a] = Scripts[a - 1]
+					end
+					Scripts[1] = Link
+					MainExists = true
+				else
+					ScriptCount = ScriptCount + 1
+					Scripts[ScriptCount] = Link
+				end
+			end
+		end
 	end
-end
+	
+	if ScriptCount > 0 then
+		local ScriptLink = Scripts[1]
+		local ScriptName, ScriptClass = ScriptLink:match("([%w-_%%]+)%.?(%l*)%.lua$")
+		ScriptName = ScriptName:gsub("Library$", "", 1):gsub("%%(%x%x)", UrlDecode)
+		local Sub = Link:sub(19)
+		local Link = Sub:gsub("^(/[^/]+/[^/]+)/tree/[^/]+", "%1", 1)
+		local LastFolder = Link:match("[^/]+$")
+		LastFolder = LastFolder:match("^RBX%-(.-)%-Library$") or LastFolder
+
+		if MainExists then
+			local Directory = LastFolder:gsub("%%(%x%x)", UrlDecode)
+			ScriptName, ScriptClass = Directory:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			if not ScriptName then ScriptName = Directory:match("^RBX%-(.-)%-Library$") or Directory end
+			if ScriptClass == "mod" or ScriptClass == "module" then TypesSpecified = true end
+		end
 
-MakeScript(ReplicatedStorage, "App/NevermoreEngine.lua")
+		if MainExists then Directory = Directory + 1 end -- :gsub("[^/]+$", "", 1) end			
+		local Count = 0
+
+		local function LocateFolder(FolderName)
+			Count = Count + 1
+			if Count > Directory then
+				Directory = Directory + 1
+				if (Parent and Parent.Name) ~= FolderName and "Modules" ~= FolderName then
+					if FolderName == "App" and Link:find("/Quenty/NevermoreEngine/", 1, true) then FolderName = "ReplicatedStorage" end
```

Comment:
Note for the source of this file, should avoid hardcoding this stuff in. OK in an installer. Bit weird in a generic installer module.